### ISSUE TITLE
Use `@github/axe-config` as config for axe

### ISSUE THREAD

**Issue by joshblack** [primer/react#2841]
We currently configure `axe` inside of Playwright through our `toHaveNoViolations` matcher: https://github.com/primer/react/blob/main/e2e/matchers/toHaveNoViolations.ts#L47. This issue is to update the configuration here to use the `@github/axe-github` package: https://www.npmjs.com/package/@github/axe-github. These rules can be passed to the `axe.configure()` call that currently exists.
